app/facades/cocktail_facade.rb

```diff
@@ -3,63 +3,69 @@ def self.retrieve_cocktail(user_id, cocktail_id)
     return false if retrieve_details(cocktail_id) == { drinks: nil }
 
     details = retrieve_details(cocktail_id)[:drinks].first
-    ingredients = get_ingredients(details)
-    measurements = get_measurements(details)
-    {
-      name: details[:strDrink],
-      thumbnail: details[:strDrinkThumb],
-      glass: details[:strGlass],
-      recipe: build_recipe(ingredients, measurements),
-      instructions: details[:strInstructions],
-      rating: find_rating(user_id, cocktail_id)
-    }
+    build_details_hash(details, user_id, cocktail_id)
   end
 
   def self.retrieve_details(cocktail_id)
-    if cocktail_id.present?
-      CocktailService.get_cocktail_details(cocktail_id)
-    else
-      CocktailService.random_cocktail
-    end
+    return CocktailService.get_cocktail_details(cocktail_id) if cocktail_id.present?
+    CocktailService.random_cocktail
   end
 
   def self.retrieve_search_results(query, user_id)
-    response = CocktailService.search_cocktails(query)
-    return false if response[:drinks].nil?
+    by_name = CocktailService.search_by_name(query)
+    by_ingredient = CocktailService.search_by_ingredient(query)
+    return false if by_name[:drinks].nil? && by_ingredient[:drinks].nil?
 
-    response[:drinks].map do |drink|
-      {
-        id: drink[:idDrink],
-        name: drink[:strDrink],
-        thumbnail: drink[:strDrinkThumb],
-        rating: find_rating(user_id, drink[:idDrink])
-      }
-    end
+    search_results = merge_results(by_name, by_ingredient)
+    build_basics_hash(search_results, user_id)
+  end
+
+  def self.merge_results(by_name, by_ingredient)
+    by_name[:drinks] = [] if by_name[:drinks].nil?
+    by_ingredient[:drinks] = [] if by_ingredient[:drinks].nil?
+    by_name[:drinks] += by_ingredient[:drinks]
   end
```

Comment:
Would love better syntax for this if anyone has ideas here

Reply:
Looks good to me...